nucleus/common/jul-extensions/src/main/java/fish/payara/jul/PayaraLogger.java

```diff
@@ -0,0 +1,524 @@
+            if (bundle != null) {
+                record.setResourceBundleName(bundle.getBaseBundleName());
+                record.setResourceBundle(bundle);
+            }
+            logOrQueue(processNow, record);
+        }
+    }
+
+
+    @Override
+    public void logrb(final Level level, final String sourceClass, final String sourceMethod,
+        final ResourceBundle bundle, final String msg, final Throwable thrown) {
+        final PayaraLoggingStatus status = getLoggingStatus();
+        if (!isLoggable(level, status)) {
+            return;
+        }
+        final boolean processNow = isReady(status);
+        if (processNow || isToQueue(status)) {
+            final LogRecord record = new EnhancedLogRecord(level, msg, false);
+            record.setLoggerName(getName());
+            record.setSourceClassName(sourceClass);
+            record.setSourceMethodName(sourceMethod);
+            record.setThrown(thrown);
+            if (bundle != null) {
+                record.setResourceBundleName(bundle.getBaseBundleName());
+                record.setResourceBundle(bundle);
+            }
+            logOrQueue(processNow, record);
+        }
+    }
+
+
+    @Override
+    public void entering(final String sourceClass, final String sourceMethod, final Object[] params) {
+        final PayaraLoggingStatus status = getLoggingStatus();
+        if (!isLoggable(Level.FINER, status)) {
+            return;
+        }
+        String msg = "ENTRY";
+        if (params == null || params.length == 0) {
+            logp(Level.FINER, sourceClass, sourceMethod, msg);
+            return;
+        }
+        final boolean processNow = isReady(status);
+        if (!processNow && !isToQueue(status)) {
+            return;
+        }
+        // ' {i}' are 4 characters per parameter. Methods with many parameters will inflate.
+        final StringBuilder template = new StringBuilder(msg.length() + 4 * params.length).append(msg);
+        for (int i = 0; i < params.length; i++) {
+            template.append(" {").append(i).append('}');
+        }
+        logp(Level.FINER, sourceClass, sourceMethod, template.toString(), params);
+    }
+
+
+    @Override
+    public void throwing(final String sourceClass, final String sourceMethod, final Throwable thrown) {
+        logp(Level.FINER, sourceClass, sourceMethod, "THROW", thrown);
+    }
+
+
+    @Override
+    public void log(final LogRecord record) {
+        final PayaraLoggingStatus status = getLoggingStatus();
+        // Logging not configured yet, don't block client, but buffer the record
+        // to be formatted on completely configured system.
+        if (isToQueue(status)) {
+            StartupQueue.getInstance().add(this, record);
+            return;
+        }
+        // Logging is configured, but now flushes buffers, we have to wait
+        // to avoid racing with buffers
+        while (status == PayaraLoggingStatus.FLUSHING_BUFFERS) {
```

Comment:
This will end in an infinitive loop.  Status is not updated within the while loop which means it is stuck forever when `status == FLUSHING_BUFFERS`

Reply:
Hmm, you are right! Seems I made some mistake and any test doesn§t cover this situation. It is not much probable, so it never happened in testing, but it is an error. Thanks a lot! I will fix it.

Reply:
and why is this check (about FLUSHING_BUFFERS) only in this method? And not in the different log() methods?

Reply:
Seems when I wrote this code it was a pretty bad day, I just remember I wanted to write test for that (and had no time for that until I forgot it) and then that I noticed after weeks or months that those bootstrap logs are somehow ... shorter. I thought that the problem is just my memory, so yet again, thanks for finding it!